### ISSUE TITLE
[Feature request] Translation pluralisation support

### ISSUE THREAD

**Issue by NiklasBr** [pimcore/pimcore#4979]
# Feature Request

## Feature description

Consider strings such as *"%number reviews"* or "*Available in %count colours"* on a product page.

If *%number* is `1` then the translated string will not make sense in English. English is pretty straight-forward like this. But other Germanic language such as German (you know this better than me) can also change definite article going from "das *Noun*" to "die *Noune*" (with an -e suffix), thus making the problem even more glaring.

Pimcore already uses [Symfony translation](https://symfony.com/doc/current/translation.html), but the interface for shared translations does not expose the ability to translate sigular/plural strings.

This is complicated by the fact that languages also differ on the zero items count, where English, German, Ductch and so pluralises the noun, but other languages such as [Polish ](http://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html#pl) and Ukrainian can have four different plural/singular forms depending in item count.

So, in summary, my feature request requires both a more expressive file format (likely the same as Symfony) for storing translation strings and a more complex UI. But I think it would be worth it long-term.

**Comment by dvesh3** [comment]
Also consider new MessageFormat from Symfony https://symfony.com/doc/4.4/translation/message_format.html#pluralization
